packages/util/src/versionDetect.ts

```diff
@@ -116,7 +118,14 @@ export function detectPackage ({ name, path, type, version }: PackageInfo, pathO
 
   entry.push({ path: getPath(path, pathOrFn), type, version });
 
-  if (entry.length !== 1) {
+  // if we have more than one entry at DIFFERENT version types then warn. If there is more than one entry at the same
+  // version and ESM/CJS dual warnings are disabled, then do not display warnings
+  const entriesSameVersion = entry.every((e) => e.version === version);
+  const esmCjsWarningDisabled = POLKADOTJS_DISABLE_ESM_CJS_WARNING_FLAG in process.env && process.env[POLKADOTJS_DISABLE_ESM_CJS_WARNING_FLAG] === '1';
```

Comment:
```suggestion
  const esmCjsWarningDisabled = xglobal.process && POLKADOTJS_DISABLE_ESM_CJS_WARNING_FLAG in xglobal.process.env && xglobal.process.env[POLKADOTJS_DISABLE_ESM_CJS_WARNING_FLAG] === '1';
```

This ensures we work in Node, browser and Deno environments. (The Deno has a CI failure)

Reply:
(PS: I would just merge in master as well, it should make it slightly easier, with the above suggestion, after https://github.com/polkadot-js/common/pull/1894)